app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -343,16 +359,21 @@ case class EvaluationResults(
           if evaluationContext.addressLookup(formComponentId.baseComponentId) || evaluationContext
             .overseasAddressLookup(formComponentId.baseComponentId) =>
         whenVisible(formComponentId) {
-          val indexedComponentId = formComponentId.modelComponentId.indexedComponentId
-          val addressAtoms: List[ModelComponentId.Atomic] =
-            if (evaluationContext.addressLookup(formComponentId.baseComponentId))
-              Address.fields(indexedComponentId).filter(_.atom != Address.uk)
-            else
-              OverseasAddress.fields(indexedComponentId).toList
-
-          val variadicValues: List[Option[VariadicValue]] = addressAtoms.map(atom => recData.variadicFormData.get(atom))
-          val addressLines = variadicValues.collect { case Some(VariadicValue.One(value)) if value.nonEmpty => value }
-          ExpressionResult.AddressResult(addressLines)
+          val modelComponentId = formComponentId.modelComponentId
+          val isModelFormComponentIdPure = modelComponentId.indexedComponentId.isPure
+          val isReferenceIndexed = evaluationContext.indexedComponentIds.exists(
+            _.baseComponentId === modelComponentId.baseComponentId
+          )
+          if (isModelFormComponentIdPure && isReferenceIndexed) {
```

Comment:
Now we are duplicating this logic in two places, can we remove the duplication?

Reply:
>. (def addressRepresentation(typeInfo: TypeInfo) need to handle possibility of a List
but it returns `List[String]` where each string is part of the address. If this method takes ListResult(List(AddressResult)) the return type of List[String] wouldn't make sense as it will contain all lines for all addresses in one List
Unless I misunderstood that.

Reply:
>if agentFullName is used only as ${bulletedList(agentFullName)} list is not being produced
didn't get that

Reply:
> if agentFullName is used only as ${bulletedList(agentFullName)} list is not being produced

do you mean that if `agentFulName` in the smart string without `bulletedList` then it is rendered as empty? I don't know how to render it tbh. Do we have any requirements on how to render list of addresses to a string?

Reply:
> but it returns List[String] where each string is part of the address. If this method takes ListResult(List(AddressResult)) the return type of List[String] wouldn't make sense as it will contain all lines for all addresses in one List
Unless I misunderstood that.

If `def addressRepresentation(...): List[String` is called on a `ListResult` of addresses we resulting list should contains individual addresses concatenated by `,<space> `. It is better default than showing nothing.